### ISSUE TITLE
check_syntax: false positive for table_line = workarea

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#2137]
```abap
REPORT zfoobar.
TYPES: BEGIN OF ty_tab,
  a TYPE string,
  b TYPE string,
  END OF ty_tab.
DATA itab TYPE TABLE OF ty_tab.
DATA wa TYPE ty_tab.
READ TABLE itab TRANSPORTING NO FIELDS WITH TABLE KEY table_line = wa.
IF sy-subrc = 0.
  WRITE 'FOUND'.
ENDIF.
```

/zfoobar.prog.abap[8, 1]: Component "table_line" not found in structure(check_syntax)

The easy workaround is to use `a = wa-a and b = wa-b`. However, the syntax above is correct and should not be an error.

**Comment by larshp** [comment]
@mbtools try again, 2.78.9
